In [1]:
# importing essential libraries
import numpy as np
import pandas as pd

In [6]:
# loading the dataset
transactions_data = pd.read_csv('../../data/transactions_clean.csv')
products_data = pd.read_csv('../../data/products.csv')

In [7]:
transactions_data.head(5)

,date,product_id,units_sold,revenue,cogs,gross_profit
0,2024-01-01,P001,15,1349.85,675.0,674.85
1,2024-01-01,P002,16,479.84,192.0,287.84
2,2024-01-01,P003,17,339.83,136.0,203.83
3,2024-01-01,P004,12,719.88,360.0,359.88
4,2024-01-01,P005,5,649.95,275.0,374.95


In [27]:
transactions_data['units_sold'].describe().loc["50%"]

np.float64(21.0)

In [32]:
def units_sold_summary(df):
    """Showing all the statistical information of units sold column"""
    
    print("Statistical summary of units sold")
    display(df['units_sold'].describe())

def mean_median_comparison(df):
    """Comparing the mean and median of units sold column to guess the distribution"""
    mean = round(df['units_sold'].mean(), 2)
    median = round(df['units_sold'].median(), 2)
    difference = round(abs(mean - median), 2)
    
    if mean > median:
        print(f"The mean value ({mean}) is higher than the median ({median}) by {difference}.")
        print(f"This indicates a right-skewed distribution pulled toward high-sales days.")
    elif median > mean:
        print(f"The median value ({median}) is higher than the mean ({mean}) by {difference}.")
        print(f"This indicates a left-skewed distribution pulled toward low-sales days.")
    else:
        print("The mean and median are equal, indicating a perfectly balanced, symmetrical distribution.")


units_sold_summary(transactions_data)
print("\n")
mean_median_comparison(transactions_data)

Statistical summary of units sold


count    1825.000000
mean       22.332603
std        10.818861
min         4.000000
25%        14.000000
50%        21.000000
75%        29.000000
max        72.000000
Name: units_sold, dtype: float64



The mean value (22.33) is higher than the median (21.0) by 1.33.
This indicates a right-skewed distribution pulled toward high-sales days.


### Summary of Units Sold & Distribution Analysis

* **Mean:** 22.33
* **Median (50%):** 21.00
* **Standard Deviation:** 10.82
* **Minimum:** 4.00
* **Maximum:** 72.00

#### Key Insights & Comparison
* **Mean vs. Median:** The mean (22.33) is higher than the median (21.00) by **1.33 units**. 
* **Distribution Shape:** Because the mean is higher than the median, the distribution is right-skewed (pulled toward higher values). While the median represents the true physical middle of the dataset, the mean is being pulled upward by occasional high-sales days.
* **Standard Deviation:** A standard deviation of 10.82 indicates that daily sales typically fluctuate by about 11 units above or below the average.

#### Typical Daily Sales Volume
While this business averages about 22 units sold per day, a typical day actually sits closer to 21 units because a few exceptionally high-sales days (peaking at 72) are pulling the average upward.

<img src="distribution_chart_a.png" width="500" alt="Right Skewed Distribution">

In [36]:
round(transactions_data['units_sold'].skew(), 2)

np.float64(0.86)

In [34]:
transactions_data['units_sold'].kurtosis()

np.float64(0.6985572279824215)

In [43]:
def shape_distribution(df):
    """Figuring out the Shapeness and symmetery of the distribution"""

    skewness = df['units_sold'].skew()
    kurtosis = df['units_sold'].kurtosis()

    print(f"The skewness ({skewness:.2f}) is between 0.5 and 1.0 which indicates that distribution is moderately skewed.")
    print(f"The kurtosis ({kurtosis:.2f}) is above 0 which indicates the distribution peak is sharper than normal.")

shape_distribution(transactions_data)

The skewness (0.86) is between 0.5 and 1.0 which indicates that distribution is moderately skewed.
The kurtosis (0.70) is above 0 which indicates the distribution peak is sharper than normal.


### Summary of Shapeness of the distribution
* **skewness**: 0.86
* **kurtosis**: 0.70

#### Key comparison and insights: 
* The skewness 0.86 in distribution indicates that the data is moderately right skewed which actually indicate that the more items that got sold per day are higher than the average meanwhile there are some days in which the sales decreases soo much. 
* The kurtosis above 0 indicates that the peak is sharper in distribution which indicates that the items sold on that quantity is really comman. 